grub-core/kern/efi/tpm.c

```diff
@@ -188,15 +188,15 @@ grub_tpm1_log_event(grub_efi_handle_t tpm_handle, unsigned char *buf,
   if (!grub_tpm_present(tpm))
     return 0;
 
-  event = grub_zalloc(sizeof (Event) + grub_strlen(description) + 1);
+  event = grub_zalloc(sizeof (TCG_PCR_EVENT) + grub_strlen(description) + 1);
   if (!event)
     return grub_error (GRUB_ERR_OUT_OF_MEMORY,
 		       N_("cannot allocate TPM event buffer"));
 
-  event->pcrindex = pcr;
-  event->eventtype = EV_IPL;
-  event->eventsize = grub_strlen(description) + 1;
-  grub_memcpy(event->event, description, event->eventsize);
+  event->PCRIndex = pcr;
+  event->EventType = EV_IPL;
+  event->EventSize = grub_strlen(description) + 1;
+  grub_memcpy(event->Event, description, event->EventSize);
 
   algorithm = TCG_ALG_SHA;
   status = efi_call_7 (tpm->log_extend_event, tpm, buf, (grub_uint64_t) size,
```

Comment:
I've tried building GRUB with `-Werror=int-conversion`, but my GCC doesn't have a problem with this.

If it fixes your build, I think that it would be more correct to explicitly cast `buf` to `grub_efi_physical_address_t` here than to change the protocol structures.  Also, you might have to do the same below for `hash_log_extend_event`.

Reply:
yes, of course. it should be `(grub_efi_physical_address_t)buf`, the same to `hash_log_extend_event`